### PR TITLE
Containerization

### PR DESCRIPTION
**Pull request by rabbitstack**
Assembling Docker images for AA and running app agent inside container. Image building can be triggered through regular Maven life cycle targets. Refer to README for more details.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,73 @@
+FROM ubuntu:latest
+LABEL maintainer="Sematext"
+
+ENV BASE_DIR /opt/spm
+ENV FPM_PATH scripts/fpm
+
+RUN \
+  apt-get update && \
+  apt-get -y install \
+          software-properties-common \
```
Comment (komljen) [inline Dockerfile]: Spacing needs to be consistent, also above I see 2 vs 4 spaces.

File: README.md [hunk 2]
```diff
@@ -40,6 +40,28 @@ To build Sematext App Agent you need:
...
+```
+
+If Docker daemon is listening on TCP socket, you can set `DOCKER_HOST` environment variable and start
+the build with regular user:
+
+```bash
+DOCKER_HOST=tcp://0.0.0.0:2375 mvn clean install dockerfile:build
+```
+
+Once image is built, launching a new container with Sematext App Agent can be achieved with the following command:
```
Comment (komljen) [inline README.md]:
```suggestion
Once the image is built, launching a new container with Sematext App Agent can be achieved with the following command:
```

File: scripts/docker/entrypoint.sh [hunk 3]
```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
+
+SPM_MONITOR_STANDALONE_CLASS="com.sematext.spm.client.StandaloneMonitorAgent"
+
+if [ -z "$JAVA" ]
```
Comment (komljen) [inline scripts/docker/entrypoint.sh]: if; then needs to be in sync, somewhere `then` is in the new line.

File: scripts/docker/unver.sh [hunk 4]
```diff
@@ -0,0 +1,13 @@
+if [ -z "$SPM_HOME"]; then
+	SPM_HOME=/opt/spm
```
Comment (komljen) [inline scripts/docker/unver.sh]: Tab vs spaces...

File: scripts/docker/unver.sh [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+fi
+
+for f in $SPM_HOME/spm-monitor/lib/*.jar; 
+do
```
Comment (komljen) [inline scripts/docker/unver.sh]: for; do ...

File: scripts/docker/entrypoint.sh [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+else
+  JAVA_OPTIONS="$JAVA_OPTIONS $JAVA_DEFAULTS"
+fi
+
+#if [[ ! "$JAVA_OPTIONS" =~ "-XX:OnOutOfMemoryError" ]];
+#then
+   #JAVA_OPTIONS="$JAVA_OPTIONS -XX:OnOutOfMemoryError=\"kill -9 %p\""
+#fi
+
+if [ -z "$SPM_MONITOR_JAR" ]; then
```
[the remark this answers is not in the document]
Comment (bsmid) [reply scripts/docker/entrypoint.sh]: Storm, Redis and HAProxy.

File: Dockerfile [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+          unzip \
+          bzip2 \
+          curl \
+	  sudo \
+	  locales && \
+  curl -skvLO \
+       -H "Cookie: oraclelicense=accept-securebackup-cookie;" \
+       http://download.oracle.com/otn-pub/java/jdk/8u191-b12/2787e4a523244c269598db4e85c51e0c/jdk-8u191-linux-x64.tar.gz && \
```
[the remark this answers is not in the document]
Comment (hakman) [reply Dockerfile]: @bsmid Let's use openjdk 11, no issues with license and works on our ES nodes.

[the remark this answers is not in the document]
Comment (hakman) [reply Dockerfile]: @bsmid Let's use openjdk 11, no issues with license and works on our ES nodes.

[the remark this answers is not in the document]
Comment (otisg) [reply Dockerfile]: +1 for OpenJDK.  Downloading Java's JDK, even 8, without agreeing to the EULA is risky.

File: Dockerfile [hunk 8]
```diff
@@ -0,0 +1,64 @@
+FROM ubuntu:18.04
+LABEL maintainer="Sematext"
+
+ENV BASE_DIR /opt/spm
+ENV FPM_PATH scripts/fpm
+
+RUN \
+  apt-get update && \
+  apt-get -y install \
+       openjdk-11-jdk \
```
Comment (hakman) [inline Dockerfile]:
```suggestion
       openjdk-11-jre-headless \
```

**Pull request merged**